code/modules/mob/living/carbon/human/human_resist.dm

```diff
@@ -9,7 +9,7 @@
 			"<span class='notice'>You stop, drop, and roll!</span>"
 			)
 		sleep(30)
-		if(fire_stacks <= 0)
+		if(fire_stacks <= 0 && !(has_modifier_of_type(/datum/modifier/fire)))
 			visible_message(
 				"<span class='danger'>[src] has successfully extinguished themselves!</span>",
 				"<span class='notice'>You extinguish yourself.</span>"
```

Comment:
Will this double up with any messages from the modifiers expiring?

Reply:
Unfortunately, yes. Though I'm not certain how to resolve that.

Reply:
You could add a boolean variable at the top of the if that remembers if they had any fire modifiers, and if they did, skip this message? i.e.,
> if(conditions)
> > var/had_mod = has_modifier_of_type(/datum/modifier/fire)
> > [...]
> > if(fire_stacks <= 0 && !had_mod && !(has_modifier of type())